### PR TITLE
Makefile: date format of log should not show signatures

### LINKED ISSUE #2834: user git configuration for signatures will affect build variables
**Description**

if user's have git configurations like `git config --global log.showSignature=true` it will cause an issue in the Makefile variable output, like:
```shell
vbatts@possibly:~/src/github.com/sigstore/cosign$ make
[288/1916] /bin/sh: 1: Syntax error: "(" unexpected
CGO_ENABLED=0 go build -trimpath -ldflags "-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=v2.0.0-63-ge71faa1a -X sigs.k8s.io/release-utils/version.gitCommit=e71faa1af0383d860481f569ef2c01694d37b00b -X sigs.k8s.io/release-utils/version.gitTreeState="clean" -X sigs.k8s.io/release-utils/version.buildDate=" -o cosign ./cmd/cosign
```

Troubleshot with running the shell in `make` as `bash -x`:
```patch
diff --git a/Makefile b/Makefile
index e8802b1b..8dc5ceb9 100644
--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+ifeq (,$(shell echo $$DEBUG))
+else
+SHELL = bash -x
+endif
+
 # Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
 ifeq (,$(shell go env GOBIN))
 GOBIN=$(shell go env GOPATH)/bin
```
```shell
vbatts@possibly:~/src/github.com/sigstore/cosign$ export DEBUG=1
vbatts@possibly:~/src/github.com/sigstore/cosign$ make                                        
+ go env GOBIN                                                                                
+ git diff --quiet                                                                            
+ '[' 1 -eq 1 ']'         
+ echo 1                                                                                      
+ find cmd -iname '*.go'                  
+ find pkg -iname '*.go'                                                                      
+ find cmd -iname '*.go'                  
+ find pkg -iname '*.go'                                                                      
+ git describe --tags --always --dirty                                                        
+ git rev-parse HEAD                                                                          
+ git log -1 --pretty=%ct                                                                     
+ git log -1 --pretty=%ct                                                                     
bash: -c: line 1: syntax error near unexpected token `('                                      
bash: -c: line 1: `date -u -d "@gpg: Signature made Fri 24 Mar 2023 09:42:02 AM EDT gpg:                using RSA key 4AEE18F83AFDEB23 gpg: Good signature from "GitHub (web-flow commit sign
ing) <noreply@github.com>" [unknown] gpg: WARNING: This key is not certified with a trusted signature! gpg:          There is no indication that the signature belongs to the owner. Primary 
key fingerprint: 5DE3 E050 9C47 EA3C F04A  42D3 4AEE 18F8 3AFD EB23 1679665322" "+%Y-%m-%dT%H:%M:%SZ" 2>/dev/null || date -u -r "gpg: Signature made Fri 24 Mar 2023 09:42:02 AM EDT gpg:    
            using RSA key 4AEE18F83AFDEB23 gpg: Good signature from "GitHub (web-flow commit signing) <noreply@github.com>" [unknown] gpg: WARNING: This key is not certified with a trusted signature! gpg:          There is no indication that the signature belongs to the owner. Primary key fingerprint: 5DE3 E050 9C47 EA3C F04A  42D3 4AEE 18F8 3AFD EB23 1679665322" "+%Y-%m-%dT%
H:%M:%SZ" 2>/dev/null || date -u "+%Y-%m-%dT%H:%M:%SZ"'                                       
CGO_ENABLED=0 go build -trimpath -ldflags "-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=v2.0.0-63-ge71faa1a-dirty -X sigs.k8s.io/release-utils/version.gitCommit=e71faa1af0383d8
60481f569ef2c01694d37b00b -X sigs.k8s.io/release-utils/version.gitTreeState="dirty" -X sigs.k8s.io/release-utils/version.buildDate=" -o cosign ./cmd/cosign                    
+ CGO_ENABLED=0                                
+ go build -trimpath -ldflags '-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=v2.0.0-63-ge71faa1a-dirty -X sigs.k8s.io/release-utils/version.gitCommit=e71faa1af0383d860481f569ef2
c01694d37b00b -X sigs.k8s.io/release-utils/version.gitTreeState=dirty -X sigs.k8s.io/release-utils/version.buildDate=' -o cosign ./cmd/cosign          
vbatts@possibly:~/src/github.com/sigstore/cosign$ git log -1 --pretty=%ct                     
gpg: Signature made Fri 24 Mar 2023 09:42:02 AM EDT                                                                                                                                          
gpg:                using RSA key 4AEE18F83AFDEB23                                                                                                                                           
gpg: Good signature from "GitHub (web-flow commit signing) <noreply@github.com>" [unknown]    
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.                                                                                                                Primary key fingerprint: 5DE3 E050 9C47 EA3C F04A  42D3 4AEE 18F8 3AFD EB23                                                                                                                  
1679665322                                    
```

**Version**

building from main e71faa1

### PR DESCRIPTION
**Pull request by vbatts**
#### Summary
if user's have git configurations like `git config --global log.showSignature=true` it will cause an issue in the Makefile variable output, like:
```shell
vbatts@possibly:~/src/github.com/sigstore/cosign$ make
[288/1916] /bin/sh: 1: Syntax error: "(" unexpected
CGO_ENABLED=0 go build -trimpath -ldflags "-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=v2.0.0-63-ge71faa1a -X sigs.k8s.io/release-utils/version.gitCommit=e71faa1af0383d860481f569ef2c01694d37b00b -X sigs.k8s.io/release-utils/version.gitTreeState="clean" -X sigs.k8s.io/release-utils/version.buildDate=" -o cosign ./cmd/cosign
```

Fixes #2834

Now it runs clean if this corner case happens:

```shell

vbatts@possibly:~/src/github.com/sigstore/cosign$ make
+ go env GOBIN
+ git diff --quiet
+ '[' 1 -eq 1 ']'
+ echo 1
+ find cmd -iname '*.go'
+ find pkg -iname '*.go'
+ find cmd -iname '*.go'
+ find pkg -iname '*.go'
+ git describe --tags --always --dirty
+ git rev-parse HEAD
+ git -c log.showSignature=false log -1 --pretty=%ct
+ git -c log.showSignature=false log -1 --pretty=%ct
+ date -u -d @1679669412 +%Y-%m-%dT%H:%M:%SZ
CGO_ENABLED=0 go build -trimpath -ldflags "-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=v2.0.0-64-gbb9af6bb-dirty -X sigs.k8s.io/release-utils/version.gitCommit=bb9af6bb1323c116e3b6b6149770aaea92ca7a95 -X sigs.k8s.io/release-utils/version.gitTreeState="dirty" -X sigs.k8s.io/release-utils/version.buildDate=2023-03-24T14:50:12Z" -o cosign ./cmd/cosign
+ CGO_ENABLED=0
+ go build -trimpath -ldflags '-buildid= -X sigs.k8s.io/release-utils/version.gitVersion=v2.0.0-64-gbb9af6bb-dirty -X sigs.k8s.io/release-utils/version.gitCommit=bb9af6bb1323c116e3b6b6149770aaea92ca7a95 -X sigs.k8s.io/release-utils/version.gitTreeState=dirty -X sigs.k8s.io/release-utils/version.buildDate=2023-03-24T14:50:12Z' -o cosign ./cmd/cosign
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -27,7 +32,7 @@ GIT_TAG ?= dirty-tag
 GIT_VERSION ?= $(shell git describe --tags --always --dirty)
 GIT_HASH ?= $(shell git rev-parse HEAD)
 DATE_FMT = +%Y-%m-%dT%H:%M:%SZ
-SOURCE_DATE_EPOCH ?= $(shell git log -1 --pretty=%ct)
+SOURCE_DATE_EPOCH ?= $(shell git -c log.showSignature=false log -1 --pretty=%ct)
```
Comment (znewman01) [inline Makefile:35]: Maybe `git log -1 --no-show-signature  --pretty=%ct`?

**Review by znewman01 — DISMISSED** [review]
CC @cpanato

File: Makefile (hunk 1 above)
Comment (cpanato) [reply Makefile:35]: +1

Comment (vbatts) [reply Makefile:35]: right on

**Comment by vbatts** [comment]
updated PTAL

**Review by znewman01 — APPROVED** [review]

**Review by cpanato — APPROVED** [review]

**Pull request merged**